metadata.yaml

```diff
@@ -0,0 +1,39 @@
+name: synapse
+display-name: Synapse
+summary: Matrix synapse chat server.
+description: |
+  A charm for the matrix synapse chat server.
+
+  The charm deploys a matrix synapse server.
+
+  Synapse is a drop in replacement for other chat servers like Mattermost and Slack.
+
+  This charm is useful if you want to spin up your own chat instance.
+
+assumes:
+  - k8s-api
+
+containers:
+  synapse:
+    resource: synapse-image
+    mounts:
+      - storage: data
+        location: /data
+
+resources:
+  synapse-image:
+    type: oci-image
+    description: OCI image for Synapse
+    upstream-source: matrixdotorg/synapse:latest
+
+storage:
+  data:
+    type: filesystem
+    location: /data
+
+requires:
+  nginx-route:
+    interface: nginx-route
+  db:
+    interface: pgsql
```

Comment:
We probably want to use the new `postgresql_client` interface (https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/postgresql_client/v0/README.md)

This should also use the lib from https://github.com/canonical/data-platform-libs.

But this represents some work and could be for another PR.

Reply:
I agree, I even have it in the README, as I noticed as well that the old interface is getting deprecated. The path I took is to reuse what we have and I know it is working (and how), swapping it out should be easy, it is some additional work like you've said. I had feedback on using `traefik` instead of `nginx-ingress-integrator` and that would be fine as well, still in terms of architecture there is no difference, it's just a design decision and a component we're swapping.